packages/devtools_app/lib/src/shared/primitives/encoding.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:vm_service/vm_service.dart';
+
+/// Encodes and decodes a value of type [T].
+abstract class EncodeDecode<T> {
+  Object toEncodable(T value);
+
+  T decode(Object value);
+
+  T? decodeNullable(Object? value) {
+    if (value == null) return null;
+    return decode(value);
+  }
+}
+
+/// Encodes and decodes HeapSnapshotGraph.
```

Comment:
put HeapSnapshotGraph in brackets []. Same for the other class names referenced in dart doc below.